cmd/print.go

```diff
@@ -244,84 +244,86 @@ func PrintProgressBarsOnEvents(w io.Writer, bus event.Bus) {
 	p := mpb.New(mpb.WithWidth(80), mpb.WithOutput(w))
 	progress := map[string]*mpb.Bar{}
 
-	// wire up a subscription to print download progress to streams
-	bus.SubscribeTypes(func(_ context.Context, e event.Event) error {
-		lock.Lock()
-		defer lock.Unlock()
-		log.Debugw("handle event", "type", e.Type, "payload", e.Payload)
-
-		switch evt := e.Payload.(type) {
-		case event.DsSaveEvent:
-			evtID := fmt.Sprintf("%s/%s", evt.Username, evt.Name)
-			cpl := int64(math.Ceil(evt.Completion * 100))
-
-			switch e.Type {
-			case event.ETDatasetSaveStarted:
-				bar, exists := progress[evtID]
-				if !exists {
-					bar = addElapsedBar(p, 100, "saving")
-					progress[evtID] = bar
+	if bus != nil {
```

Comment:
I would rather have the guard on the entry point next to `shouldColorOutput` and make it the caller responsibility to ensure the bus is not nil. Or at least make this log out an `event bus is nil` message so we don't swallow this error state in the future.